packages/server/src/index.ts

```diff
@@ -1652,20 +1656,20 @@ export class App {
                     const endingNodeData = endingNode.data
                     if (!endingNodeData) return res.status(500).send(`Ending node ${endingNode.id} data not found`)
 
-                    if (endingNodeData && endingNodeData.category !== 'Chains' && endingNodeData.category !== 'Agents') {
-                        return res.status(500).send(`Ending node must be either a Chain or Agent`)
-                    }
+                    const isEndingNode = Object.values(endingNodeData.outputs ?? {}).includes('EndingNode')
```

Comment:
To solve the double run issue, we can move the isEndingNode logic to the top of the init function and then reuse the variable down here:
```
  const isEndingNode = nodeData?.outputs?.output === 'EndingNode'
  if (isEndingNode && !options.isRun) return

```

Reply:
so there are 2 ways to solve the issue right, one is we do it at the `CustomFunction.ts` like my comment above, or the way u suggest where we do it at the `server/index.ts`

Reply:
Oops... I put this in the wrong place... I'll move this comment to CustomFunction.ts